### PR TITLE
Rename resources to suit the resource limits context.

### PR DESCRIPTION
**Pull request by kaniyan**
The package and the spring properties pertaining to `resource-limits`  are currently identified with the prefix `plan` and this has been renamed with a more suited variant (`resourcelimits`).

- The package has been renamed from `plan` to `resourcelimits`.
- The spring property prefix has been renamed from `hono.plan.prometheus-based` to `hono.resource-limits.prometheus-based`.
- Also the documentation has been updated.

### CODE REVIEW

**Comment by ctron** [comment]
I prefer the new naming over the existing one. But I would also understand to keep the current one, as we are close to the release. Changing this after 1.0.0 however would break compatibility.

I would lean towards renaming now.

**Comment by sophokles73** [comment]
I agree with @ctron, in particular since I believe that these properties are not widely used yet ...

**Comment by kaniyan** [comment]
@ctron @sophokles73  It is the same motivation for me to raise this PR before 1.0.0 release. I agree that it is better to have this renamed before the 1.0.0 release.

File: site/documentation/content/admin-guide/amqp-adapter-config.md [hunk 1]
```diff
@@ -137,11 +137,11 @@ The following table provides an overview of the configuration variables and corr
...
+| `HONO_PLAN_PROMETHEUS_BASED_HOST`<br>`--hono.resourceLimits.prometheusBased.host` | no | none | The host name or IP address of the Prometheus server to retrieve the metrics data from. This property needs to be set in order to enable the Prometheus based checks. |
+| `HONO_PLAN_PROMETHEUS_BASED_PORT`<br>`--hono.resourceLimits.prometheusBased.port` | no | `9090` | The port of the Prometheus server to retrieve metrics data from. |
+| `HONO_PLAN_PROMETHEUS_BASED_CACHE_MIN_SIZE`<br>`--hono.resourceLimits.prometheusBased.cacheMinSize` | no | `20` | The minimum size of the cache to store the metrics data retrieved from the Prometheus server. The cache is used for storing the current amount of data exchanged with devices of tenants. |
+| `HONO_PLAN_PROMETHEUS_BASED_CACHE_MAX_SIZE`<br>`--hono.resourceLimits.prometheusBased.cacheMaxSize` | no | `1000` | The maximum size of the cache to store the metrics data retrieved from the Prometheus server. |
+| `HONO_PLAN_PROMETHEUS_BASED_CACHE_TIMEOUT`<br>`--hono.resourceLimits.prometheusBased.cacheTimeout` | no | `600` | The number of seconds after which the cached metrics data should be considered invalid. |
```
Comment (sophokles73) [inline site/documentation/content/admin-guide/amqp-adapter-config.md]: the environment variable names need to be adapted as well ...

File: site/documentation/content/admin-guide/http-adapter-config.md [hunk 2]
```diff
@@ -139,11 +139,11 @@ The following table provides an overview of the configuration variables and corr
...
+| `HONO_PLAN_PROMETHEUS_BASED_HOST`<br>`--hono.resourceLimits.prometheusBased.host` | no | none | The host name or IP address of the Prometheus server to retrieve the metrics data from. This property needs to be set in order to enable the Prometheus based checks. |
+| `HONO_PLAN_PROMETHEUS_BASED_PORT`<br>`--hono.resourceLimits.prometheusBased.port` | no | `9090` | The port of the Prometheus server to retrieve metrics data from. |
+| `HONO_PLAN_PROMETHEUS_BASED_CACHE_MIN_SIZE`<br>`--hono.resourceLimits.prometheusBased.cacheMinSize` | no | `20` | The minimum size of the cache to store the metrics data retrieved from the Prometheus server. The cache is used for storing the current amount of data exchanged with devices of tenants. |
+| `HONO_PLAN_PROMETHEUS_BASED_CACHE_MAX_SIZE`<br>`--hono.resourceLimits.prometheusBased.cacheMaxSize` | no | `1000` | The maximum size of the cache to store the metrics data retrieved from the Prometheus server. |
+| `HONO_PLAN_PROMETHEUS_BASED_CACHE_TIMEOUT`<br>`--hono.resourceLimits.prometheusBased.cacheTimeout` | no | `600` | The number of seconds after which the cached metrics data should be considered invalid. |
```
Comment (sophokles73) [inline site/documentation/content/admin-guide/http-adapter-config.md]: the environment variable names need to be adapted as well ...

File: site/documentation/content/admin-guide/kura-adapter-config.md [hunk 3]
```diff
@@ -140,11 +140,11 @@ The following table provides an overview of the configuration variables and corr
...
+| `HONO_PLAN_PROMETHEUS_BASED_HOST`<br>`--hono.resourceLimits.prometheusBased.host` | no | none | The host name or IP address of the Prometheus server to retrieve the metrics data from. This property needs to be set in order to enable the Prometheus based checks. |
+| `HONO_PLAN_PROMETHEUS_BASED_PORT`<br>`--hono.resourceLimits.prometheusBased.port` | no | `9090` | The port of the Prometheus server to retrieve metrics data from. |
+| `HONO_PLAN_PROMETHEUS_BASED_CACHE_MIN_SIZE`<br>`--hono.resourceLimits.prometheusBased.cacheMinSize` | no | `20` | The minimum size of the cache to store the metrics data retrieved from the Prometheus server. The cache is used for storing the current amount of data exchanged with devices of tenants. |
+| `HONO_PLAN_PROMETHEUS_BASED_CACHE_MAX_SIZE`<br>`--hono.resourceLimits.prometheusBased.cacheMaxSize` | no | `1000` | The maximum size of the cache to store the metrics data retrieved from the Prometheus server. |
+| `HONO_PLAN_PROMETHEUS_BASED_CACHE_TIMEOUT`<br>`--hono.resourceLimits.prometheusBased.cacheTimeout` | no | `600` | The number of seconds after which the cached metrics data should be considered invalid. |
```
Comment (sophokles73) [inline site/documentation/content/admin-guide/kura-adapter-config.md]: the environment variable names need to be adapted as well ...

File: site/documentation/content/admin-guide/mqtt-adapter-config.md [hunk 4]
```diff
@@ -139,11 +139,11 @@ The following table provides an overview of the configuration variables and corr
...
+| `HONO_PLAN_PROMETHEUS_BASED_HOST`<br>`--hono.resourceLimits.prometheusBased.host` | no | none | The host name or IP address of the Prometheus server to retrieve the metrics data from. This property needs to be set in order to enable the Prometheus based checks. |
+| `HONO_PLAN_PROMETHEUS_BASED_PORT`<br>`--hono.resourceLimits.prometheusBased.port` | no | `9090` | The port of the Prometheus server to retrieve metrics data from. |
+| `HONO_PLAN_PROMETHEUS_BASED_CACHE_MIN_SIZE`<br>`--hono.resourceLimits.prometheusBased.cacheMinSize` | no | `20` | The minimum size of the cache to store the metrics data retrieved from the Prometheus server. The cache is used for storing the current amount of data exchanged with devices of tenants. |
+| `HONO_PLAN_PROMETHEUS_BASED_CACHE_MAX_SIZE`<br>`--hono.resourceLimits.prometheusBased.cacheMaxSize` | no | `1000` | The maximum size of the cache to store the metrics data retrieved from the Prometheus server. |
+| `HONO_PLAN_PROMETHEUS_BASED_CACHE_TIMEOUT`<br>`--hono.resourceLimits.prometheusBased.cacheTimeout` | no | `600` | The number of seconds after which the cached metrics data should be considered invalid. |
```
Comment (sophokles73) [inline site/documentation/content/admin-guide/mqtt-adapter-config.md]: the environment variable names need to be adapted as well ...

Comment (kaniyan) [reply site/documentation/content/admin-guide/mqtt-adapter-config.md]: @sophokles73 I have adapted them and pushed the changes. Pls have a look.

**Comment by sophokles73** [comment]
@kaniyan can you please also document this change in the release notes under *API Changes*?

**Comment by sophokles73** [comment]
@kaniyan any news on this?

**Comment by kaniyan** [comment]
@sophokles73 I am working on it and after that I will push the changes.

**Comment by kaniyan** [comment]
@sophokles73 I have pushed the changes. Please take a look.

**Pull request merged**